flink-tests/src/test/java/org/apache/flink/test/checkpointing/UnalignedCheckpointITCase.java

```diff
@@ -99,87 +105,162 @@
 
-    private static UnalignedSettings createPipelineSettings(
-            int parallelism, int slotsPerTaskManager, boolean slotSharing) {
-        return createPipelineSettings(parallelism, slotsPerTaskManager, slotSharing, 0);
-    }
+                final SingleOutputStreamOperator<Long> deduplicated =
+                        combinedSource
+                                .flatMap(new SourceAwareMinEmittingFunction(NUM_SOURCES))
+                                .name("min")
+                                .uid("min")
+                                .slotSharingGroup(slotSharing ? "default" : "min");
+                addFailingPipeline(minCheckpoints, slotSharing, deduplicated);
+            }
+        };
 
-    private static UnalignedSettings createPipelineSettings(
-            int parallelism, int slotsPerTaskManager, boolean slotSharing, int timeout) {
-        int numShuffles = 4;
-        return new UnalignedSettings(UnalignedCheckpointITCase::createPipeline)
-                .setParallelism(parallelism)
-                .setSlotSharing(slotSharing)
-                .setNumSlots(slotSharing ? parallelism : parallelism * numShuffles)
-                .setSlotsPerTaskManager(slotsPerTaskManager)
-                .setExpectedFailures(5)
-                .setFailuresAfterSourceFinishes(1)
-                .setAlignmentTimeout(timeout);
+        @Override
+        public String toString() {
+            return name().toLowerCase();
+        }
     }
 
-    private static UnalignedSettings createCogroupSettings(int parallelism) {
-        int numShuffles = 10;
-        return new UnalignedSettings(UnalignedCheckpointITCase::createMultipleInputTopology)
-                .setParallelism(parallelism)
-                .setSlotSharing(true)
-                .setNumSlots(parallelism * numShuffles)
-                .setSlotsPerTaskManager(parallelism)
-                .setExpectedFailures(5)
-                .setFailuresAfterSourceFinishes(1);
+    @Parameterized.Parameters(name = "{0} with {2} channels, p = {1}, timeout = {3}")
+    public static Object[][] parameters() {
+        Object[] defaults = {Topology.PIPELINE, 1, MIXED, 0};
+
+        return Stream.of(
+                        new Object[] {Topology.PIPELINE, 1, LOCAL},
+                        new Object[] {Topology.PIPELINE, 1, REMOTE},
+                        new Object[] {Topology.PIPELINE, 5, LOCAL},
+                        new Object[] {Topology.PIPELINE, 5, REMOTE},
+                        new Object[] {Topology.PIPELINE, 20},
+                        new Object[] {Topology.PIPELINE, 20, MIXED, 1},
+                        new Object[] {Topology.MULTI_INPUT, 5},
+                        new Object[] {Topology.MULTI_INPUT, 10},
+                        new Object[] {Topology.UNION, 5},
+                        new Object[] {Topology.UNION, 10})
+                .map(params -> addDefaults(params, defaults))
+                .toArray(Object[][]::new);
     }
```

Comment:
@pnowojski has a valid point: if we comment out all but one, `Stream.of` goes from producing an `Object[]` to `Object`. 
I pulled out the parameters and it's better readable now, makes it possible what Piotr wants, and we have a trailing `,`, so that we can comment out any test easily.

Reply:
I see... good point!